codec-http2/src/main/java/io/netty/handler/codec/http2/HttpToHttp2ConnectionHandler.java

```diff
@@ -61,27 +68,92 @@ private int getStreamId(HttpHeaders httpHeaders) throws Exception {
      */
     @Override
     public void write(ChannelHandlerContext ctx, Object msg, ChannelPromise promise) {
-        if (msg instanceof FullHttpMessage) {
-            FullHttpMessage httpMsg = (FullHttpMessage) msg;
-            boolean hasData = httpMsg.content().isReadable();
-            boolean httpMsgNeedRelease = true;
+
+        if (msg instanceof HttpMessage) {
```

Comment:
Just to aid in readability, can you separate these blocks out into separate methods (e.g. `writeHttpMessage` and `writeHttpContent`)?